### PR TITLE
Drop all "theories/*/vo.itarget" files and compute the corresponding information automatically.

### PR DESCRIPTION
**Pull request by ghost**
At the moment, whenever a new `*.v` file is added/renamed/removed to/in/from the `theories`, we have to update the corresponding `vo.itarget` files.

I propose that we drop all those `vo.itarget` files and let `make` simply discover them automatically.

### CODE REVIEW

**Comment by maximedenes** [comment]
I told Matej offline that generated files need specific treatment and helped him implement a fix he'll push soon.

**Comment by matejkosik** [comment]
Closing this pull-request and reopen it against `trunk`.

**Comment by Zimmi48** [comment]
You don't need to open a new pull request to change the base branch.

**Comment by Zimmi48** [comment]
You can edit the base branch (edit button located next to the title) and force-push a rebased version in the same branch.

**Comment by matejkosik** [comment]
Great! Thanks.

File: Makefile.common [hunk 1]
```diff
@@ -147,14 +147,16 @@ LINKCMX:=$(CORECMA:.cma=.cmxa) $(STATICPLUGINS:.cmo=.cmx)
...
+GENVOFILES := $(GENVFILES:.v=.vo)
 
-PLUGINSVO:=  $(foreach f, $(wildcard plugins/*/vo.itarget), \
-                $(addprefix $(dir $(f)),$(shell cat $(f))))
+THEORIESVO := $(patsubst %.v,%.vo,$(shell find theories -type f -name "*.v")) \
+              $(filter theories/%, $(GENVOFILES))
 
-ALLVO:= $(THEORIESVO) $(PLUGINSVO)
-VFILES:= $(ALLVO:.vo=.v)
+PLUGINSVO := $(patsubst %.v,%.vo,$(shell find plugins -type f -name "*.v" | grep -v plugins/micromega/MExtraction.v)) \
```
Comment (matejkosik) [inline Makefile.common]: This looks ugly, but if no other `*.v` files need to be filtered out, maybe it can stay like this.
If it turns out that we want to filter out more than one file, it may be worth to define "blacklist".

Comment (maximedenes) [inline Makefile.common]: This certainly cannot stay like that. Can you say more about why this file has to be filtered out? And if it has, why it is there?

Comment (maximedenes) [inline Makefile.common]: I don't see any occurrence of MExtraction in the entire codebase. So was it included by mistake in the repo?

Comment (matejkosik) [inline Makefile.common]: Hmm, isn't [this](https://github.com/coq/coq/blob/trunk/plugins/micromega/MExtraction.v) it?

Comment (maximedenes) [inline Makefile.common]: I don't understand, what do you mean?

Comment (matejkosik) [inline Makefile.common]: The `vo.itarget` file in `trunk` does not try to trigger its compilation ...

Comment (maximedenes) [inline Makefile.common]: Yes, that's my point. So I guess removing it from the repo would not change anything. No file mentions this one.

Comment (matejkosik) [inline Makefile.common]: I'll do that.

**Comment by matejkosik** [comment]
The change above will be later squashed with the rest.

At the moment I am investigating the differences between the sets of installed files.

```
> lib/coq/theories/Logic/PropFacts.vo
952a954
> lib/coq/theories/Logic/.coq-native/NCoq_Logic_PropFacts.cmi
988a991
> lib/coq/theories/Logic/.coq-native/NCoq_Logic_PropFacts.cmxs
1011a1015
> lib/coq/theories/Logic/.coq-native/NCoq_Logic_PropFacts.o
1079a1084
> lib/coq/theories/Logic/.coq-native/NCoq_Logic_PropFacts.cmx
1094a1100
> lib/coq/theories/Logic/PropFacts.v
1100a1107
> lib/coq/theories/Logic/PropFacts.glob
```
whether they are desired or not ...

**Comment by maximedenes** [comment]
I think I'm missing something. You say difference, but you show only one list.

**Comment by matejkosik** [comment]
When I checked whether
- before the changes in the `Makefile.common`
- after the changes in the `Makefile.common`

file, I have found out that after my changes, a few more files are compiled and installed:
- `lib/coq/theories/Logic/PropFacts.vo`
- `lib/coq/theories/Logic/.coq-native/NCoq_Logic_PropFacts.cmi`
- `lib/coq/theories/Logic/.coq-native/NCoq_Logic_PropFacts.cmxs`
- `lib/coq/theories/Logic/.coq-native/NCoq_Logic_PropFacts.o`
- `lib/coq/theories/Logic/.coq-native/NCoq_Logic_PropFacts.cmx`
- `lib/coq/theories/Logic/PropFacts.v`
- `lib/coq/theories/Logic/PropFacts.glob`

The reason is that, originally, `theories/Logic/PropFacts.v` file is part of the git repository, but this file is (originally) neither compiled nor installed.

In my branch, this file gets automatically compiled and installed (together with other things).

The question is, what do we want to do? If we want to simplify the `Makefile.common` as I propose, we can 
- either let that file be compiled and installed (if that makes sense)
- or could remove it from the git repository (if it does not make sense to give it to the users)
- or we could keep it in the git repository but introduce some kind of "blacklist" mechanism so that that file, like in `trunk`, will be neither compiled nor installed.

What do we want to do with it?

**Comment by maximedenes** [comment]
Oh! Now I understand. You've discovered a file that was never compiled nor installed in the stdlib. That's another sign that this PR is useful :) `PropFacts.v` was introduced by the following commit:

```
commit 4bd5dcfeb558f826d90123357a3e5336f96b2213
Author: Hugo Herbelin <Hugo.Herbelin@inria.fr>
Date:   Tue Sep 22 18:45:27 2015 +0200

    Adding an amazing property of Prop.
```

@herbelin What should we do with this file? Make it be compiled and installed?

**Comment by letouzey** [comment]
Hum,  have you taken a few second to look at the content of `MExtraction.v` before obliviating it ? 

Agreed, it is not meant to be compiled nor installed at each build of Coq. But that doesn't mean that this file is useless : it is in fact the *real* source for file `micromega.ml`. Hence it could be interesting to have it around if the micromega plugin evolves someday, or if the extraction makes huge progress. Both are relatively unlikely, but we never know. Okay, we could do some git archeology to recover this file in this event, but that's clearly not developper-friendly.  Here I would rather suggest migrating this file somewhere outside the `$(shell find ...)`, for instance in `dev`. But odds are great that it will be forgotten there forever.

To sum up, the `PropFacts` example above may show the interest of this PR, but the `MExtraction` example shows on the contrary the constraints this PR imposes.

**Comment by maximedenes** [comment]
I still think that the constraint of making sure we compile all `.v` files of the archive is very sane. In the case of micromega, I would be tempted to try to actually generate `micromega.ml` from `MExtraction` when building. Btw, today, nothing makes it clear that we should not edit the `micromega.ml` file.

**Comment by herbelin** [comment]
@matejkosik: as said by @letouzey and @maximedenes, it should have been in vo.itarget. My bad.

**Comment by matejkosik** [comment]
@herbelin : OK, I'll make sure that `theories/Logic/PropFacts.v` is compiled and installed.

@letouzey : in that case (since `plugins/micromega/micromega.ml{i,}` can be generated from `plugins/micromega/MExtraction.v` (which I think this is great!)) the simplest and the least unusual thing to do is:
- remove `plugins/micromega/micromega.ml{i,}` from git
- make sure that `plugins/micromega/micromega.ml{i,}` is generated from `plugins/micromega/MExtraction.v`
- add `plugins/micromega/micromega.ml{i,}` to `.gitignore`
- check if all (what we currently manage to track) is still compilable
- check whether the `plugins/micromega/micromega.ml` file contains anything that is not part of the generated file (modulo reformatting) (after briefely looking at the files, they seem to differ only in formatting)

No?

**Comment by matejkosik** [comment]
This is a followup of PR #515 .

**Comment by matejkosik** [comment]
[Travis results](https://travis-ci.org/matejkosik/coq/builds/238391476) are all green.
